### PR TITLE
fix: trampoline unwraps

### PR DESCRIPTION
**Pull request by ruben-arts**
Fixes:
```
pixi g install -e pixi35 "pixi=0.35" --expose pixi0.35=pixi
pixi0.35
BIG PANIC
```

TODO:
- [x] : Add ci build trampolines to the repo.

# Acknowledgement
I (ruben) am not a fan of the size increase I introduced. Anyone that can figure out how to cleanly simplify and shrink the binary should feel free to give it a go!
- We don't _need_ `miette` I just want less scary errors than a `panic`
- We don't need the full `pixi_utils` we could split up the crate into one that does the executable logic.

### CODE REVIEW

**Comment by nichmor** [comment]
what is the size increase of binary after adding miette?

**Comment by ruben-arts** [comment]
I don't know yet @nichmor

File: crates/pixi_trampoline/src/main.rs [hunk 1]
```diff
@@ -18,39 +20,40 @@ struct Metadata {
...
 
-fn prepend_path(extra_path: &str) -> String {
-    let path = env::var("PATH").unwrap();
+fn prepend_path(extra_path: &str) -> miette::Result<String> {
+    let path = env::var("PATH").into_diagnostic().wrap_err("Couldn't get 'PATH'")?;
     let mut split_path = env::split_paths(&path).collect::<Vec<_>>();
     split_path.insert(0, PathBuf::from(extra_path));
-    let new_path = env::join_paths(split_path).unwrap();
-    new_path.to_string_lossy().into_owned()
+    let new_path = env::join_paths(split_path).into_diagnostic().wrap_err("Couldn't join PATH's")?;
```
Comment (nichmor) [inline crates/pixi_trampoline/src/main.rs:39]: maybe we could also format split_path in the error message

**Comment by ruben-arts** [comment]
Before:
```
❯ du -ha . | sort -k 2
1.3M    .
192K    ./pixi-trampoline-aarch64-apple-darwin.zst
136K    ./pixi-trampoline-aarch64-pc-windows-msvc.exe.zst
260K    ./pixi-trampoline-aarch64-unknown-linux-musl.zst
204K    ./pixi-trampoline-x86_64-apple-darwin.zst
144K    ./pixi-trampoline-x86_64-pc-windows-msvc.exe.zst
332K    ./pixi-trampoline-x86_64-unknown-linux-musl.zst
```

After:
```
❯ du -ha | sort -k 2
1.8M	.
284K	./pixi-trampoline-aarch64-apple-darwin.zst
224K	./pixi-trampoline-aarch64-pc-windows-msvc.exe.zst
364K	./pixi-trampoline-aarch64-unknown-linux-musl.zst
292K	./pixi-trampoline-x86_64-apple-darwin.zst
240K	./pixi-trampoline-x86_64-pc-windows-msvc.exe.zst
384K	./pixi-trampoline-x86_64-unknown-linux-musl.zst
```

Significantly bigger but not enough to make me want to put time in making it smaller for now.

File: crates/pixi_trampoline/src/main.rs (hunk 1 above)
Comment (ruben-arts) [reply crates/pixi_trampoline/src/main.rs]: Done

**Review by nichmor — APPROVED** [review]
looks good!

**Pull request merged**